### PR TITLE
Inspect: Simplify for users to attach sanitized Grafana data to issues

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR was originally created during one of our internal hack days. As a Grafanista on issue triage, you often find yourself in a situation where the information reported by creator of an issue is limited. This PR adds a `Bug report` tab to the Panel Inspector so we can help users attach essential information used by us when we investigate issues.

https://user-images.githubusercontent.com/562238/121995674-9fd98f80-cda7-11eb-90f9-ce9d7b2d9ead.mp4

The sanitize part can be further improved (see issue) but there is enough there to classify it as an `alpha feature`. 

The thing I struggle with is what to call this and where to place it? Maybe this belongs to the Share Panel modal instead/too?

Also, the information text could use some help from someone with better English. Can you help me out with that @achatterjee-grafana?

**Which issue(s) this PR fixes**:
Fixes #32532

**Special notes for your reviewer**:

### CODE REVIEW

**Review by mckn — APPROVED** [review]
LGTM! Awesome work with this!

**Comment by hugohaggmark** [comment]
> LGTM! Awesome work with this!

Yes, it was a great collaboration!

**Comment by marefr** [comment]
Regarding the naming, would _Report a bug_ or _Bug report_  work better than than _Share_? Or in what other circumstances would you be interested in this feature when not reporting a bug?

**Comment by hugohaggmark** [comment]
> Regarding the naming, would _Report a bug_ or _Bug report_ work better than than _Share_? Or in what other circumstances would you be interested in this feature when not reporting a bug?

Thanks, I think that makes a lot of sense except the user can't report the bug using this feature but I see what you mean. I'll try it out.

**Comment by hugohaggmark** [comment]
Using `Bug report` as suggested by @marefr and updated recording.

**Comment by hugohaggmark** [comment]
My plan is to present this in the next UX feedback session, before merging it :)

File: public/app/features/inspector/InspectBugReportTab.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+          />
+        </Field>
+        <div className={styles.featureBadge}>
+          <FeatureBadge featureState={FeatureState.alpha} />
+        </div>
+      </div>
+      <HorizontalGroup>
+        <Button type="button" icon="download-alt" onClick={onDownloadClick} variant="secondary">
+          Download collected data
+        </Button>
```
Comment (dprokop) [inline public/app/features/inspector/InspectBugReportTab.tsx:100]: I was surprised when this downloaded an md file. I was expecting a zip file with dashboard.json, panel.json etc files. 

Do you think we need this button at all? I can imagine the workflow here to be:
1. I want to report an issue.
2. I go to inspect and copy the md.
3. I paste it into the issue.

Hard for me to think about a case when I would download the fie for ltr use.

File: public/app/features/dashboard/components/Inspector/utils.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+export function inspectPackager(): Packager {
+  return {
+    package: function (items: CollectorItem[]): string {
+      const markdowns: string[] = [];
+      const newline = '\n';
+
+      markdowns.push(`#### Shared data from Grafana (${new Date().toISOString()})`);
+      for (const item of items) {
+        let markdown = '';
+        markdown += `##### ${item.name}  `;
```
Comment (dprokop) [inline public/app/features/dashboard/components/Inspector/utils.ts:18]: FYI, OS and browser details were empty when I tried this

**Review by dprokop — APPROVED** [review]
This is already a huge improvement in the shape that it currently is. Huge thumbs up for the effort @hugohaggmark.

File: public/app/features/inspector/InspectBugReportTab.tsx (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/inspector/InspectBugReportTab.tsx:100]: Wow, that is an excellent idea!! ❤️

File: public/app/features/dashboard/components/Inspector/utils.ts (hunk 2 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/components/Inspector/utils.ts:18]: Did you get build errors? Needed to add a new package for this functionality.

File: public/app/features/inspector/InspectBugReportTab.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+              } else {
+                setSanitizers([]);
+              }
+            }}
+          />
+        </Field>
+        <div className={styles.featureBadge}>
+          <FeatureBadge featureState={FeatureState.alpha} />
+        </div>
+      </div>
```
[the remark this answers is not in the document]
Comment (hugohaggmark) [reply public/app/features/inspector/InspectBugReportTab.tsx:96]: Agree, thank you!

File: public/app/features/dashboard/components/Inspector/utils.ts (hunk 2 above)
Comment (dprokop) [reply public/app/features/dashboard/components/Inspector/utils.ts:18]: Of course, you are right - only now I have realized there is a TS error in my console. nvm

File: public/app/features/inspector/InspectBugReportTab.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dprokop) [reply public/app/features/inspector/InspectBugReportTab.tsx:96]: Or maybe even better - move it to a Bug report tab to draw attention🤔 ?

**Comment by dprokop** [comment]
One last thing I have noticed - the bug report tab is empty for panels that do not query data, i.e. news panel or text panel.

File: public/app/features/inspector/InspectBugReportTab.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (hugohaggmark) [reply public/app/features/inspector/InspectBugReportTab.tsx:96]: Yepp, sounds like a plan

**Comment by hugohaggmark** [comment]
> One last thing I have noticed - the bug report tab is empty for panels that do not query data, i.e. news panel or text panel.

Of course, I always forget those. I'll have a look 👍

**Review by marefr — APPROVED** [review]
LGTM

**Comment by hugohaggmark** [comment]
Closing this for now to limit my open PR WIP.

**Pull request closed without merging**